src/rpc/rawtransaction.cpp

```diff
@@ -333,6 +334,25 @@ void TxToJSON(const CTransaction& tx, const uint256 hashBlock, UniValue& entry)
             }
         }
     }
+
+    auto feePaid = tx.GetFeePaid([&](const COutPoint& out) -> std::optional<CAmount> {
```

Comment:
What's the difference between this, and `auto feePaid = cacheCoins.GetValueIn(tx) - tx.GetValueOut();` where `cacheCoins` is a `CCoinsViewCache`?

Reply:
This is interesting. Do we have examples of where we use `CCoinsViewCache` directly from the RPC methods? I'll do some investigating.

Reply:
Okay, I decided to try this, and here's the issue:
```
coins.cpp:962: const CTxOut &CCoinsViewCache::GetOutputFor(const CTxIn &) const: Assertion `coins && coins->IsAvailable(input.prevout.n)' failed.
```

`GetValueIn` is apparently only intended to be called in contexts where you're certain that the node has the prevouts available; this must be why it's on `CCoinsViewCache` and not `CCoinsView`. The partiality of the closure passed to `GetFeePaid` allows us to make the computation in a best-effort fashion, and consults more possible sources of data than `GetValueIn`. The upshot of this is (a) separately, we should make `GetValueIn` return a `std::optional` result, and (b) maybe then reuse it as part of the implementation of this closure.